tools/generator/GenBase.cs

```diff
@@ -529,20 +530,30 @@ bool CanMethodBeIsStyleSetter (Method m)
 			return GetAllDerivedInterfaces ().All (iface => !iface.Properties.Any (p => p.Name == "Is" + m.PropertyName && p.Setter == null));
 		}
 
-		// Keep sorted alphabetically
-		static readonly string[]    ObjectRequiresNew       = new string[]{
-			"Handle",
-		};
-
-		static readonly string[]    ThrowableRequiresNew    = new string []{
-			"Message",
-		};
+		static readonly HashSet<string>         ObjectRequiresNew       = new HashSet<string>(
+			typeof (object)
+				.GetMethods ()
+				.Where (m => !m.Attributes.HasFlag (MethodAttributes.SpecialName) &&
+						!m.Attributes.HasFlag (MethodAttributes.RTSpecialName))
+				.Select (m => m.Name)
+				.Concat (new[] { "Handle" }),
+			StringComparer.OrdinalIgnoreCase);
+
+		static readonly HashSet<string>         ThrowableRequiresNew    = new HashSet<string>(
+			typeof (System.Exception)
+				.GetMethods ()
+				.Where (m => !m.Attributes.HasFlag (MethodAttributes.SpecialName) &&
+						!m.Attributes.HasFlag (MethodAttributes.RTSpecialName))
+				.Select (m => m.Name)
+				.Concat (typeof (System.Exception).GetProperties ().Select (p => p.Name))
+				.Concat (new[] { "Handle" }),
+			StringComparer.OrdinalIgnoreCase);
```

Comment:
Should this be `IgnoreCase`? Can you have an `equals` and an `Equals` method?

Reply:
For CLS compliance (is that still a thing?), members need to be case-insensitive.  For example, having `Equals()`, `equals()`, and `eQuals()` methods would be perfectly valid C# and Java, but would be a CLS violation.

Since we're aiming for .NET bindings, case insensitivity for bindings thus seems appropriate.

Which makes the case for *some* form of case insensitivity.

Then there's `.IgnoreCase` vs. `.OrdinalIgnoreCase`.  In the case of Java bindings, it probably doesn't matter, but *in general* `.IgnoreCase` is locale-aware, while `.OrdinalIgnoreCase` is US-english predominant (and also matches the Windows filesystem).  One classic example where this matters is [the Turkish 'i' Problem](https://en.wikipedia.org/wiki/Dotted_and_dotless_I): in Turkey, `'i'.ToUpper()` is `İ`, not `I`.  *In general* use of `.IgnoreCase` -- or any other locale-sensitive comparison -- needs to be carefully thought out, as it could result in different things happening in different cultures.

In some places, this is desirable, e.g. sorting of strings within a UI.  For *bindings* and things that don't directly interact with a "user", locale-aware sorting is often *not* desirable.